volatility3/framework/plugins/windows/amcache.py

```diff
@@ -1,651 +1,23 @@
-            if isinstance(entry.last_modify_time, datetime.datetime):
-                yield f"Amcache: {entry.entry_type} {entry.path} registry key modified", timeliner.TimeLinerType.MODIFIED, entry.last_modify_time
-            if isinstance(entry.last_modify_time_2, datetime.datetime):
-                yield f"Amcache: {entry.entry_type} {entry.path} STANDARD_INFORMATION create time", timeliner.TimeLinerType.CREATED, entry.last_modify_time_2
-            if isinstance(entry.install_time, datetime.datetime):
-                yield f"Amcache: {entry.entry_type} {entry.path} installed", timeliner.TimeLinerType.CREATED, entry.install_time
-            if isinstance(entry.compile_time, datetime.datetime):
-                yield f"Amcache: {entry.entry_type} {entry.path} compiled (PE metadata)", timeliner.TimeLinerType.MODIFIED, entry.compile_time
-
-    @classmethod
-    def get_amcache_hive(
-        cls,
-        context: interfaces.context.ContextInterface,
-        config_path: str,
-        kernel_module_name: str,
-    ) -> Optional[registry.RegistryHive]:
-        """Retrieves the `Amcache.hve` registry hive from the kernel module, if it can be located."""
-        return next(
-            hivelist.HiveList.list_hives(
-                context=context,
-                base_config_path=interfaces.configuration.path_join(
-                    config_path, "hivelist"
-                ),
-                kernel_module_name=kernel_module_name,
-                filter_string="amcache",
-            ),
-            None,
-        )
-
-    @classmethod
-    def parse_file_key(
-        cls, file_key: reg_extensions.CM_KEY_NODE
-    ) -> Iterator[Tuple[NullableString, _AmcacheEntry]]:
-        """Parses File entries from the Windows 8 `Root\\File` key.
-
-        :param programs_key: The `Root\\File` registry key.
-
-        :return: An iterator of tuples, where the first member is the program ID string for
-        correlating `Root\\Program` entries, and the second member is the `AmcacheEntry`.
-        """
-
-        val_enum = Win8FileValName
-
-        wanted_values = [key.value for key in val_enum]
-
-        for file_entry_key in itertools.chain(
-            *(key.get_subkeys() for key in file_key.get_subkeys())
-        ):
-            vollog.debug(f"Checking Win8 File key {file_entry_key.get_name()}")
-            values = {
-                str(value.get_name()): value
-                for value in file_entry_key.get_values()
-                if value.get_name() in wanted_values
-            }
-
-            program_id = _get_string_value(values, val_enum.ProgramID.value)
-            path = _get_string_value(values, val_enum.Path.value)
-            company = _get_string_value(values, val_enum.Company.value)
-            last_mod_time = _get_datetime_filetime_value(
-                values, val_enum.LastModTime.value
+    def __getattr__(self, *args, **kwargs):
+        warnings.warn(
+            DeprecationWarning(
```

Comment:
`DeprecationWarning`s only get seen by developers who run python with a special flag, I think it needs to be `FutureWarning` (see [here](https://docs.python.org/3/library/exceptions.html#FutureWarning) for more information).

Reply:
@ikelos So I checked again and FutureWarning does print the message if I change it to override `run` instead of `__getattr__`, but I think you had suggested getattr so people would see it if they imported the file and used a classmethod rather than calling the plugin directly.

Reply:
using `__getattribute__` instead of getattr works for the command line, but still not when I import one in the windows directory, i.e. in registry.cachedump, do `from volatility3.plugins.windows import hashdump` instead of the one in windows.registry.